### PR TITLE
test-bot: hard link generated bottle file to HOMEBREW_CACHE

### PR DESCRIPTION
**Pull request by xu-cheng**

### CODE REVIEW

**Comment by mikemcquaid** [comment]
If we do this I think it's worth deleting it afterwards on cleanup too.

**Comment by xu-cheng** [comment]
There is already [cleanup code](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/cmd/test-bot.rb#L1020-Lundefined) for CI. Do you want to cleanup these files for users as well?

**Comment by mikemcquaid** [comment]
> There is already cleanup code for CI. Do you want to cleanup these files for users as well(which is somehow covered by brew cleanup)?

That's not being run in CI unless the option is passed? As-is this could cause issues if e.g. a future PR tries to download that bottle as a dependency (e.g. without `--force`) and then checksum may be incorrect depending on the bottle upload ordering.

**Comment by xu-cheng** [comment]
Updated.

File: Library/Homebrew/cmd/test-bot.rb [hunk 1]
```diff
@@ -1017,7 +1018,14 @@ def test_bot
...
     end
   ensure
-    HOMEBREW_CACHE.children.each(&:rmtree) if ARGV.include? "--clean-cache"
+    if ARGV.include? "--clean-cache"
+      HOMEBREW_CACHE.children.each(&:rmtree)
+    else
+      Dir.glob("*.bottle*.tar.gz") do |bottle_file|
+        FileUtils.rm_f HOMEBREW_CACHE/bottle_file
+      end
+    end
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/test-bot.rb]: I don't think we want this to delete all bottles in this case? That will mean it redownloads bottles used as dependencies that haven't been updated every time. Feels nicer to store and delete the specific file.

Comment (xu-cheng) [inline Library/Homebrew/cmd/test-bot.rb]: This will only delete new generated bottles. Noticed that it's scanning on current directory not HOMEBREW_CACHE.

Comment (mikemcquaid) [inline Library/Homebrew/cmd/test-bot.rb]: Aha, I see! Nice work :+1:

**Comment by mikemcquaid** [comment]
:+1:

**Pull request closed without merging**